web/thesauruses/javascript/ECMAScript 2023/strings.json

```diff
@@ -0,0 +1,352 @@
+      },
+      "concatenate_two_strings": {
+        "name": "Concatenate/join two strings",
+        "code": [
+            "str + str2",
+            "str.concat(str2)"
+        ]
+      },
+      "concatenate_many_strings": {
+        "name": "Concatenate/join many strings",
+        "code": [
+            "str + str2 + str3 + str4",
+            "str.concat(str2, str3, str4)"
+        ]
+      },
+      "is_all_alphabetical": {
+        "name": "Is string all alphabetical characters?",
+        "code": "(/^[a-zA-Z]+$/g).test(str)"
+      },
+      "is_all_numerical": {
+        "name": "Is string all numerical characters?",
+        "code": "(/^[0-9]+$/g).test(str)"
+      },
+      "is_all_alphanumeric": {
+        "name": "Is string all alphanumeric characters?",
+        "code": "(/^[a-zA-Z0-9]+$/g).test(str)"
+      },
+      "is_decimal": {
+        "name": "Is string a decimal number?",
+        "code": "(/^[0-9]+\\.[0-9]+$/g).test(str)"
+      },
+      "is_all_whitespaces": {
+        "name": "Is string all whitespace characters?",
+        "code": "(/^[\\s]+$/g).test(str)"
+      },
+      "is_all_uppercase": {
+        "name": "Is string all uppercase characters?",
+        "code": "str === str.toUpperCase()"
+      },
+      "is_all_lowercase": {
+        "name": "Is string all lowercase characters?",
+        "code": "str === str.toLowerCase()"
+      },
+      "is_in_titlecase": {
+        "name": "Is string formatted in title case?",
+        "code": "str === str.replace(/\\w+/g, (w) => w[0].toUpperCase() + w.slice(1).toLowerCase())"
+      },
+      "does_substring_exist": {
+        "name": "Does a substring exist in a string?",
+        "code": [
+            "str.includes(substring)",
+            "str.indexOf(substring) !== -1"
+        ]
+      },
+      "find_start_index_of_substring": {
+        "name": "Find index of where a substring starts",
+        "code": [
+            "str.indexOf(substring)",
+            "str.search(substring)"
+        ]
+      },
+      "find_start_index_of_additional_substring": {
+        "name": "Find index of an additional substring (or starting at another index)",
+        "code": [
+            "str.indexOf(substring, index)",
+            "str.search(substring)"
+        ]
+      },
+      "find_start_index_of_substring_from_end": {
+        "name": "Find substring index starting at end",
+        "code": [
+            "str.lastIndexOf(substring)",
+            "str.search(substring)"
+        ]
+      },
+      "count_occurrences_of_substring": {
+        "name": "Find number of occurences of a substring",
+        "code": [
+            "str.split(substring).length - 1",
+            "str.match(/substring/g).length"
+        ]
+      },
+      "get_leftmost_characters": {
+        "name": "Get a specified number of characters from the left",
+        "code": [
+            "str.slice(start, end)",
+            "str.substring(start, end)",
+            "str.substr(start, length)"
```

Comment:
According to MDN, they say substr() is deprecated and not to use it anymore. (I'm not sure when it got deprecated, so it may or may not exist in 2009/2020/2021/2023.) It should probably be removed in 2021/2023 at least.